packages/router/src/__tests__/links.test.tsx

```diff
@@ -39,6 +40,107 @@ describe('<NavLink />', () => {
     expect(getByText(/Dunder Mifflin/)).toHaveClass('activeTest')
   })
 
+  it('receives active class on the same pathname with search parameters', () => {
+    const mockLocation = createDummyLocation(
+      '/search-params',
+      '?tab=main&page=1'
+    )
+
+    const { getByText } = render(
+      <LocationProvider location={mockLocation}>
+        <NavLink
+          activeClassName="activeTest"
+          to={`/search-params?page=1&tab=main`}
+        >
+          Dunder Mifflin
+        </NavLink>
+      </LocationProvider>
+    )
+
+    expect(getByText(/Dunder Mifflin/)).toHaveClass('activeTest')
+  })
+
+  it('receives active class on the same pathname only', () => {
+    const mockLocation = createDummyLocation('/pathname', '?tab=main&page=1')
+
+    const { getByText } = render(
+      <LocationProvider location={mockLocation}>
+        <NavLink
+          activeClassName="activeTest"
+          to={`/pathname?tab=second&page=2`}
+          activeMatchParams={[]}
+        >
+          Dunder Mifflin
+        </NavLink>
+      </LocationProvider>
+    )
+
+    expect(getByText(/Dunder Mifflin/)).toHaveClass('activeTest')
+  })
+
+  it('receives active class on the same pathname with a matched param key', () => {
+    const mockLocation = createDummyLocation(
+      '/pathname-params',
+      '?tab=main&page=1'
+    )
+
+    const { getByText } = render(
+      <LocationProvider location={mockLocation}>
+        <NavLink
+          activeClassName="activeTest"
+          to={`/pathname-params?tab=main&page=2`}
+          activeMatchParams={['tab']}
+        >
+          Dunder Mifflin
+        </NavLink>
+      </LocationProvider>
+    )
+
+    expect(getByText(/Dunder Mifflin/)).toHaveClass('activeTest')
+  })
+
+  it('receives active class on the same pathname with a matched key-value param', () => {
+    const mockLocation = createDummyLocation(
+      '/search-params',
+      '?tab=main&page=1'
+    )
+
+    const { getByText } = render(
+      <LocationProvider location={mockLocation}>
+        <NavLink
+          activeClassName="activeTest"
+          to={`/search-params?page=1&tab=main`}
+          activeMatchParams={[{ page: 1 }]}
+        >
+          Dunder Mifflin
+        </NavLink>
+      </LocationProvider>
+    )
+
+    expect(getByText(/Dunder Mifflin/)).toHaveClass('activeTest')
+  })
+
+  it('receives active class on the same pathname with a matched param key and a matched key-value param', () => {
+    const mockLocation = createDummyLocation(
+      '/search-params',
+      '?tab=main&page=1&category=book'
+    )
+
+    const { getByText } = render(
+      <LocationProvider location={mockLocation}>
+        <NavLink
+          activeClassName="activeTest"
+          to={`/search-params?page=3&tab=main&category=book`}
+          activeMatchParams={[{ category: 'book' }, 'page']}
```

Comment:
Can you add another key-value param to test against here? Or create another test that has at least two key-value and at least one simple string. Just want to make sure we show that it's `[{ foo: 'foo' }, { bar: 'bar' }]` (and not `[{ foo: 'foo', bar: 'bar' }]`)

Reply:
I added test cases for both `[{ foo: 'foo' }, { bar: 'bar' }]` and `[{ foo: 'foo', bar: 'bar' }]` as it is supported